linux/deploy_vm/deploy_vm_from_ova.yml

```diff
@@ -51,88 +49,76 @@
-            current_power_state: "{{ vm_power_state_get }}"
-        - include_tasks: ../../common/vm_get_guest_info.yml
+    - name: "Collect OVA configs"
+      include_tasks: "../../common/collect_ovf_vm_config.yml"
+      vars:
+        ovf_vm_hardware_config_path: "{{ current_test_log_folder }}"
 
-        # Try to get OS type from guest info
-        - include_tasks: get_os_type_from_guestinfo.yml
-      when: ova_guest_os_type == 'unknown'
+    - name: "Try to get OS type from guest info"
+      include_tasks: get_ova_guest_os_type.yml
 
-    # Revert to snapshot "FreshDeployedFromOVA" or "AfterUpgradeHwv" to
+    # Revert to snapshot "FreshDeployedFromOVA" to
     # proceed the following tasks so that VM's reconfiguration can be
     # performed at the first time boot.
-    - include_tasks: ../../common/vm_revert_snapshot.yml
+    - name: "Revert to snapshot '{{ base_snapshot_for_reconfig }}' for VM reconfig"
+      include_tasks: ../../common/vm_revert_snapshot.yml
       vars:
         snapshot_name: "{{ base_snapshot_for_reconfig }}"
-      when:
-        - current_power_state is defined
-        - current_power_state == "poweredOn"
 
-    # Configure username/password, ssh public key, sshd, etc in guest
-    # Reconfig Amazon Linux VM
-    - include_tasks: amazonlinux/reconfigure_amazonlinux_vm.yml
-      when: ova_guest_os_type == 'amazonlinux'
+    # Upgrade VM hardware version
+    # Note:
+    # Known issue on Ubuntu cloud image OVA deployed VM, after upgrade hardware version,
+    # VM will hang during booting, tracked in this issue:
+    # https://bugs.launchpad.net/cloud-images/+bug/1898871
+    - name: "Upgrade VM's hardware version to {{ hardware_version }}"
+      include_tasks: upgrade_ova_vm_hwv.yml
+      when:
+        - hardware_version is defined
+        - (hardware_version == "latest" or
+           (vm_hardware_version_num | int < hardware_version | int))
 
-    # Reconfig VMware Photon OS VM
-    - include_tasks: photon/reconfigure_photon_vm.yml
-      when: ova_guest_os_type == 'photon'
+    - name: "Reconfigure VM with cloud-init"
+      include_tasks: reconfigure_vm_with_cloudinit.yml
+      when: ova_guest_os_type in ['photon', 'ubuntu', 'amazon']
 
-    # Reconfig Flatcar VM
-    - include_tasks: flatcar/reconfigure_flatcar_vm.yml
+    - name: "Reconfigure Flatcar VM with Ignition"
+      include_tasks: flatcar/reconfigure_flatcar_vm.yml
       when: ova_guest_os_type == 'flatcar'
 
-    # Reconfig Ubuntu VM
-    - include_tasks: ubuntu/reconfigure_ubuntu_ova_vm.yml
-      when: ova_guest_os_type == 'ubuntu'
-
-    # Warning about other OS OVAs which are not supported
-    - ansible.builtin.debug:
-        msg: "WARN: OVA guest OS type is not supported in this project. The following tests might fail."
+    - name: "Warning about unknown guest OS type"
+      ansible.builtin.debug:
+        msg: >-
+          WARNING: The guest OS type of the OVA is not supported in this project.
```

Comment:
I think here can be changed to "not matched known guest OS type reconfiguration file/method, please add it if needed or following tests might fail."

Reply:
Message is updated.